shell/src/main/java/alluxio/master/MasterHealthCheckClient.java

```diff
@@ -199,24 +189,18 @@ public void run() {
         while (true) {
           List<InetSocketAddress> addresses = client.getMasterRpcAddresses();
           for (InetSocketAddress address : addresses) {
-            String host = address.getHostName();
-            int port = address.getPort();
-            LOG.debug("Master health check on node {}", host);
-            String cmd = String.format("ssh %s %s %s", ShellUtils.COMMON_SSH_OPTS, host,
-                "ps -ef | grep \"" + mAlluxioMasterName + "$\" | "
-                + "grep \"java\" | "
-                + "awk '{ print $2; }'");
-            LOG.debug("Executing: {}", cmd);
-            String output = ShellUtils.execCommand("bash", "-c", cmd);
-            if (output.isEmpty()) {
-              throw new IllegalStateException(
-                  String.format("Master process is not running on the host %s", host));
-            } else if (output.contains("Connection refused")) {
-              throw new IllegalStateException(
-                  String.format("Connection refused while connecting to the host %s on port %d",
-                      host, port));
+            LOG.debug("Master health check on node {}", address.getHostName());
+            try (Socket socket = new Socket()) {
```

Comment:
Why we need to make sure all master of HA members are available ?
For now, the standby master don't start rpc service, and they are not necessary to start other masters while start a specific master

Reply:
The standby masters don't start rpc service. But standby masters start up with the port. That indicates the service status of standby masters.

Reply:
@ljl1988com AFAIK, only primary master start and listen rpc port, the standby masters didn't start rpc port.

even, if i start only two master of three master, the started two master will not pass the monitor for cannot connect to the third not started master?

Reply:
@maobaolong I make sure that the standby masters start rpc port in zk and embedded mode. But the standby masters don't service. If there is one master which is down, the check will not pass.

I will add a new checker which monitor port. Don't replace the monitor proc.

Reply:
@ljl1988com 
> I make sure that the standby masters start rpc port in zk and embedded mode

you mean the rpc port which default 19998? Could you give some arguments or details to prove the standby masters start and listen the rpc(default 19998) port?

May be you can use netstat anp|grep 19998

Or just paste some related codes.

Reply:
Yes. I use netstat anp | grep 19998. I find the rpc port exists. So I make usre that the standby masters start rpc port.
 
https://github.com/Alluxio/alluxio/blob/4da0071c805923489d9da506b58cc829b82cfe71/core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java#L188-L214

`startRejectingServers` will start the rpc port.

https://github.com/Alluxio/alluxio/blob/4da0071c805923489d9da506b58cc829b82cfe71/core/server/common/src/main/java/alluxio/master/MasterProcess.java#L181-L190

https://github.com/Alluxio/alluxio/blob/4da0071c805923489d9da506b58cc829b82cfe71/core/common/src/main/java/alluxio/network/RejectingServer.java#L44-L62

PS: I also thought standby servers didn't start rpc port before i started this issue.

Reply:
The `RejectingServer` will close all connections on the socket, so I wonder if your `if(socket.isConnected())` check will return true?

Reply:
I test the checker returns true.